atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -90,24 +90,63 @@ public static SerializableTransactionManager create(
             Schema schema,
             Environment env,
             boolean allowHiddenTableAccess) {
-        return create(config, ImmutableSet.of(schema), env, allowHiddenTableAccess);
+        return create(config, Optional.absent(), ImmutableSet.of(schema), env, allowHiddenTableAccess);
+    }
+
+    /**
+     * Create a {@link SerializableTransactionManager} with provided configuration,
+     * {@link SSLSocketFactory}, {@link Schema}, and an environment in which to
+     * register HTTP server endpoints.
+     *
+     * This method is deprecated in favor of specifying the ssl configuration for
+     * client-side leader connections via {@link LeaderConfig}'s sslConfiguration
+     * and not through the sslSocketFactory argument.
```

Comment:
I think it's reasonable that the security definition for connecting to the leaders, timestamp and lock services can occur outside of the rather large AtlasDb config block, and that it'd be better for this, as a library, to require the application owners to capture and define an appropriate SSLContext for talking to those leaders.